src/sidebar/reducers/test/links-test.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+var links = require('../links');
+
+var init   = links.init;
+var update = links.update.UPDATE_LINKS;
+var action = links.actions.updateLinks;
+
+describe('sidebar.reducers.links', function() {
+  describe('#init()', function() {
+    it('returns a null links object', function() {
+      assert.deepEqual(init(), {links: null});
+    });
+  });
+
+  describe('#update.UPDATE_LINKS()', function() {
+    it('returns the given newLinks as the links object', function() {
+      assert.deepEqual(
+        update('CURRENT_STATE', {newLinks: 'NEW_LINKS'}),
```

Comment:
I agree that tests partially serve as documentation. But here what I'm attempting to document is that what the `UPDATE_LINKS()` actually does is blindly take whatever value is passed to it (as `newLinks: <VALUE>`) and return it  (as `links: <VALUE>`). The actual value, `'NEW_LINKS'` in this test, is opaque as far as `UPDATE_LINKS()` is concerned. It could be a boolean, an object, whatever, and `UPDATE_LINKS()` would work the same. The fact that what `serviceUrl` actually passes happens to be an object is not `UPDATE_LINKS()`'s concern.

I could use an object here instead of a string in order to be consistent with what `serviceUrl` actually passes here. But then if `serviceUrl` were changed to store a different type of object these `links` test would not be failing, but would be out of date. In a weak way, the readability of the `links` tests would have been coupled to the `serviceUrl` code, whereas I'd like them to be coupled only to the `links` code.

Similarly the current value which `annotationUI` (or whatever part of our Redux machinery) passes to `UPDATE_LINKS()`, `'CURRENT_STATE'` in this test, is also an opaque value (`UPDATE_LINKS()` actually doesn't touch it at all).

I agree that it's unfortunate that, because the tests use strings here, from reading the tests it might look as if the links reducer is only for storing strings when in fact it's for storing any kind of value and in practice is currently only used to store an object. This is case where in Python you would [use a sentinel object](https://www.seanh.cc/posts/sentinel) rather than a string as a clear way to say this to the reader that this is an opaque value. I don't think any such thing exists in JavaScript so I feel back on all caps strings hoping that would be clear enough.

Even though currently it's only used for storing the links templates, this links reducer could easily be generalised into a "store one opaque value in Redux an be able to update that value by completely replacing it with a new value" reducer.

Do you think that makes sense?